### DISCUSSION TITLE
Drop support for the Xhyve hypervisor

### DISCUSSION THREAD

**Discussion by jacob-carlborg** [cross-platform-actions/action#79] [Ideas]
I'm thinking about dropping support for the Xhyve hypervisor. It seems to cause more problems than it's worth. In general, it seems to be less stable than QEMU. There have been several issues (#77, #72, #61) reported about the VM freezes, all of them seem to be due to Xhyve being used and most likely due to the same root cause. It has proven very difficult to get to the root cause of these issues. Even if I would manage to find the root cause it might not be fixed. The Xhyve repository hasn't been updated in [three years](https://github.com/machyve/xhyve/commits/master/). There have been other issues related to building Xhyve, like manually having to add entitlements. The latest thing that has recently happened is that the package has been disabled in Homebrew due to not compiling anymore.

So what do you think, should support for Xhyve be dropped?

**Comment by manxorist** [comment]
I am testing the fix for #73 right now. If that works, I am solely using QEMU and do not really care about xhyve any more.

**Reply by manxorist** [reply]
Well, this was not successful, I am seeing user space crashes inside OpenBSD which did never happen with Xhyve instead of QEMU. So it looks like OpenBSD itself is unstable inside QEMU.

**Reply by manxorist** [reply]
The second run worked, I will re-run it a couple of times.

**Reply by jacob-carlborg** [reply]
> Well, this was not successful, I am seeing user space crashes inside OpenBSD which did never happen with Xhyve instead of QEMU. So it looks like OpenBSD itself is unstable inside QEMU.

Could you please file an issue and include a link to the failing job? Might be possible to tweak the flags passed to QEMU.

**Reply by manxorist** [reply]
> Could you please file an issue and include a link to the failing job? Might be possible to tweak the flags passed to QEMU.

#80

**Reply by jacob-carlborg** [reply]
Now when https://github.com/cross-platform-actions/action/issues/47 is fixed, this might not be a problem anymore? Or at least less of a problem.

**Reply by manxorist** [reply]
I have had 22 runs with QEMU on Ubuntu so far (FreeBSD 14.0, FreeBSD 13.2, OpenBSD 7.4, NetBSD 9.3), all without any problem. Looks like at least I do not need Xhyve any more.
